compiler/rustc_borrowck/src/region_infer/mod.rs

```diff
@@ -758,27 +758,14 @@ impl<'tcx> RegionInferenceContext<'tcx> {
         // Otherwise, we need to find the minimum remaining choice, if
         // any, and take that.
         debug!("choice_regions remaining are {:#?}", choice_regions);
-        let min = |r1: ty::RegionVid, r2: ty::RegionVid| -> Option<ty::RegionVid> {
-            let r1_outlives_r2 = self.universal_region_relations.outlives(r1, r2);
-            let r2_outlives_r1 = self.universal_region_relations.outlives(r2, r1);
-            match (r1_outlives_r2, r2_outlives_r1) {
-                (true, true) => Some(r1.min(r2)),
-                (true, false) => Some(r2),
-                (false, true) => Some(r1),
-                (false, false) => None,
-            }
+        let Some((_, &min_choice)) = choice_regions.iter().enumerate().find(|(i, &r1)| {
+            choice_regions.iter().enumerate().all(|(j, &r2)| {
+                *i == j || self.universal_region_relations.outlives(r1, r2)
```

Comment:
```suggestion
                self.universal_region_relations.outlives(r2, r1)
```
The `min_choice` should be _outlived by_ all others, not _outlives_ all others. I'll be worried if we don't have tests that tripped for this.

There's also no need to check `i == j` because universal_region_relations is reflexive, so `outlives(r1, r1) == true`.

Reply:
I just found a local test failed for my change:

```rust
async fn async_ret_impl_trait2<'a, 'b>(a: &'a u8, b: &'b u8) -> impl Trait<'a>
where
    'b: 'a,
{
    (a, b)
}
```

Will fix it.